components/brave_ads/browser/ads_notification_handler.cc

```diff
@@ -120,4 +157,47 @@ const void* AdsNotificationHandler::UserDataKey() {
   return &kUserDataKey;
 }
 
+void AdsNotificationHandler::OperationCompleted(
+    const std::string& notification_id) {
+
+  auto iter = pending_callbacks_.find(notification_id);
+  std::move(iter->second).Run();
+  pending_callbacks_.erase(iter);
+#if defined(OS_ANDROID)
+  StartShutDownTimerIfNecessary(notification_id);
+#endif
+}
+
+#if defined(OS_ANDROID)
+bool AdsNotificationHandler::IsHeadless(){
+  auto state = base::android::ApplicationStatusListener::GetState();
+  bool uiless = (state ==
+      base::android::ApplicationState::APPLICATION_STATE_UNKNOWN ||
+      state ==
+      base::android::ApplicationState::APPLICATION_STATE_HAS_DESTROYED_ACTIVITIES);
+  return uiless;
+}
+
+void AdsNotificationHandler::StartShutDownTimerIfNecessary (
```

Comment:
I think the way this is usually phrased in Chromium source is `MaybeStartShutdownTimer`.

Reply:
`IfNecessary` seems to be popular too.